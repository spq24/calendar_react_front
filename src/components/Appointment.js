import React from 'react';
import { Link } from 'react-router-dom';
import { formatDate } from '../utils/format';
import PropTypes from 'prop-types'
import $ from 'jquery';

export default class Appointment extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            appointment: props.appointment
        }
    }

    static propTypes = {
        appointment: PropTypes.object.isRequired
    }

    static defaultProps = {
        appointment: {}
    }

    componentDidMount () {
        if(this.props.match) {
            $.ajax({
                type: "GET",
                url: `http://localhost:3001/appointments/${this.props.match.params.id}`,
                dataType: "JSON",
                headers: JSON.parse(sessionStorage.user)
            }).done((data) => {
                this.setState({appointment: data});
            })
        }
    }

    render () {
        return (
            <div className='appointment'>
                <Link to={`/appointments/${this.state.appointment.id}`} >
                    <h3>{this.state.appointment.name}</h3>
                </Link>
                <p>{formatDate(this.state.appointment.time)}</p>
                <Link to={`/appointments/${this.state.appointment.id}/edit`} >
                    <h4>edit</h4>
                </Link>
            </div>
        )
    }
}
