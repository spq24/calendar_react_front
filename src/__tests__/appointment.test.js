import React from 'react';
import ReactDOM from 'react-dom';
import Appointment from '../components/Appointment';
import { BrowserRouter as Router } from 'react-router-dom';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  shallow(<Appointment />);
});

describe('render', () => {
    it('should display the appointment title', () => {
    const appointment = mount(<Router><Appointment appointment={{id:1, title: 'Team standup meeting', appt_time: new Date()}} /></Router>);
    const title = <h3>Team standup meeting</h3>;
        expect(appointment.contains(title)).toEqual(true);
    })

    it('should display the appointment time', () => {
    const appointment = mount(<Router><Appointment appointment={{id:1, title: 'Team standup meeting', appt_time: new Date('04/11/2017, 12:00:00')}} /></Router>);
    const appt_time = <p>April 11 2017, 12:00:00 pm</p>;
        expect(appointment.contains(appt_time)).toEqual(true);
    })

    it('renders appointment correctly', () =>{
      const appointment = renderer.create(<Router><Appointment appointment={{id:1, title: 'Team standup meeting', appt_time: new Date('04/11/2017, 12:00:00')}} /></Router>);
        let tree = appointment.toJSON();
        expect(tree).toMatchSnapshot();
    })
});
