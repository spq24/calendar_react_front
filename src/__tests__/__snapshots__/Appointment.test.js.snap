exports[`render renders appointment correctly 1`] = `
<div
  className="appointment">
  <a
    href="/appointments/1"
    onClick={[Function]}>
    <h3>
      Team standup meeting
    </h3>
  </a>
  <p>
    April 11 2017, 12:00:00 pm
  </p>
  <a
    href="/appointments/1/edit"
    onClick={[Function]}>
    Edit
  </a>
</div>
`;
